pkg/ccrm/ccrm_cloudlet.go

```diff
@@ -40,65 +43,40 @@ var (
 	}
 )
 
-func (s *CCRMHandler) cloudletChanged(ctx context.Context, old *edgeproto.Cloudlet, in *edgeproto.Cloudlet) {
-	log.SpanLog(ctx, log.DebugLevelApi, "cloudletChanged", "cloudlet", in)
+// ApplyCloudlet implements a GRPC CloudletPlatform server method
+func (s *CCRMHandler) ApplyCloudlet(in *edgeproto.Cloudlet, stream edgeproto.CloudletPlatformAPI_ApplyCloudletServer) error {
+	ctx := stream.Context()
+	log.SpanLog(ctx, log.DebugLevelApi, "ApplyCloudlet", "cloudlet", in)
 
 	s.updateCloudletNodes(ctx, in)
 
-	var ackState edgeproto.TrackedState
-	var errState edgeproto.TrackedState
-	var successState edgeproto.TrackedState
-	var workFunc func(context.Context, *edgeproto.Cloudlet, platform.Platform, edgeproto.CacheUpdateCallback) error
+	responseSender := edgeproto.NewCloudletInfoSendUpdater(ctx, stream, in.Key)
 
-	if in.OnboardingState == edgeproto.TrackedState_CREATE_REQUESTED {
-		ackState = edgeproto.TrackedState_CREATING
-		errState = edgeproto.TrackedState_CREATE_ERROR
-		successState = edgeproto.TrackedState_READY
+	var workFunc func(context.Context, *edgeproto.Cloudlet, platform.Platform, *edgeproto.CloudletInfoSendUpdater) error
```

Comment:
might be better to define with worker function type, so the variable declaration is cleaner looking, but this is a nit.